### PR TITLE
chore(query-performance): Properly deprecate Team object fields

### PR DESCRIPTION
**Pull request by macobo**
Team lookups seem to take up the majority of time on postgres according to pganalyze. We're returning redundant data in queries as well.

This change avoids reading deprecated columns alltogether whenever querying teams, speeding up the query somewhat.

### CODE REVIEW

**Review by thmsobrmlr — APPROVED** [review]
LGTM 🚢 

Q: The fields are deprecated for almost two years now and I can't find any usage for them in code. Might be time to completely remove them?

File: posthog/models/team/team.py [hunk 1]
```diff
@@ -41,6 +41,9 @@
 
 
 class TeamManager(models.Manager):
+    def get_queryset(self):
```
Comment (pauldambra) [inline posthog/models/team/team.py:44]: Oh, this is pretty cool!

I bet dashboards and insights would benefit from this too! 🙌 🤔

**Review by pauldambra — APPROVED** [review]
:shipit:

**Comment by macobo** [comment]
> Q: The fields are deprecated for almost two years now and I can't find any usage for them in code. Might be time to completely remove them?

Problem: Self-hosted upgrades and non-rollbackable migrations (which this would be). So if users upgrade to a release that has these columns and then roll back to one that doesn't, that results in a broken instance as queries will start failing. :(

**Pull request merged**